compiler/rustc_codegen_llvm/src/builder.rs

```diff
@@ -561,6 +561,24 @@ impl<'a, 'll, 'tcx> BuilderMethods<'a, 'tcx> for Builder<'a, 'll, 'tcx> {
         count: u64,
         dest: PlaceRef<'tcx, &'ll Value>,
     ) -> Self {
+        if let OperandValue::Pair(mut v1, mut v2) = cg_elem.val && count < 1024 {
+            v1 = self.from_immediate(v1);
+            v2 = self.from_immediate(v2);
+            let ty = self.cx().val_ty(v1);
+            // Create a vector of size 2*count and store it in one instruction
+            if ty == self.cx().val_ty(v2) {
+                let count = count * 2;
+                let vec = unsafe { llvm::LLVMGetUndef(self.type_vector(ty, count as u64)) };
+                let vec = (0..count as usize).fold(vec, |acc, x| {
+                    let elt = [v1, v2][x % 2];
+                    self.insert_element(acc, elt, self.cx.const_i32(x as i32))
```

Comment:
No idea if this is better, but you could do it in O(1) LLVM instructions by making a `<2 x _>` (with two insert_elements) and then repeating that one with a shuffle like
```llvm
shufflevector <2 x i8> %x, <2 x i8> undef, <64 x i32> <i32 0, i32 1, i32 0, i32 1, i32 0, i32 1, i32 0, i32 1, i32 0, i32 1, i32 0, i32 1, i32 0, i32 1, i32 0, i32 1, i32 0, i32 1, i32 0, i32 1, i32 0, i32 1, i32 0, i32 1, i32 0, i32 1, i32 0, i32 1, i32 0, i32 1, i32 0, i32 1, i32 0, i32 1, i32 0, i32 1, i32 0, i32 1, i32 0, i32 1, i32 0, i32 1, i32 0, i32 1, i32 0, i32 1, i32 0, i32 1, i32 0, i32 1, i32 0, i32 1, i32 0, i32 1, i32 0, i32 1, i32 0, i32 1, i32 0, i32 1, i32 0, i32 1, i32 0, i32 1>
```
https://llvm.godbolt.org/z/5feKaEo5z